lib/logstash/inputs/beats/message_listener.rb

```diff
@@ -31,7 +31,9 @@ def onNewMessage(ctx, message)
       hash = message.getData
       ip_address = ip_address(ctx)
 
-      hash['@metadata']['ip_address'] = ip_address unless ip_address.nil? || hash['@metadata'].nil?
+      unless ip_address.nil? || hash['@metadata'].nil?
+        set_nested(hash, @input.field_hostip, ip_address)
+      end
       target_field = extract_target_field(hash)
 
       extract_tls_peer(hash, ctx)
```

Comment:
I understand your intention to keep explicit what's done for the various `ecs_compatibility` levels, but this replicate the behavior and we miss the meaning of `ecs_select`. I thought that the spirit of `ecs_select` is to keep in one place the mapping correspondence between ecs `:disabled` and `:v1`, the same way of using it is also present in `logstash-input-udp` and I think that keep the same style across plugins would be helpful.

I've some doubts about `['@metadata']['tls_peer']` proposed suggestions:
- could we change the type of a field when we rename it? For example "verified/unverified" for `:disabled` ECS becomes `true/false` when `:v1` is enabled. This change would require more changes in user's pipelines during the transition. 
- the `tls` section is described in [docs](https://www.elastic.co/guide/en/ecs/current/ecs-tls.html) and doesn't specify any `verified` field, so could we add a custom named field nesting into an already ECS defined structure? Do we risk conflicts with future versions? Could be this confusing?

Reply:
:+1: believe there's a misunderstanding here.
my original suggestion to review `hash['@metadata']['tls_peer']` in the context of unifying the `[input][beats]` prefix was simply to avoid a non-namespaced hash - which we can not achieve if `['@metadata']['tls_peer']['status']` has to stay.

if we need to keep it than I am less concerned about the need to change the rest but I now see that it can still provide some value if users extract the whole `[@metadata][input][beats][tls]` hash they get ECS compliant TLS information. 

> we miss the meaning of `ecs_select`

true in terms of consistency - still it hurt my :eyes: in terms of readability as there are many fields we set at once ...
not sure it's improving the code here but I am willing to :peace_symbol: with it.

> the tls section is described in docs and doesn't specify any `verified` field, so could we add a custom named field nesting into an already ECS defined structure?

was fully aware `[tls][verified]` not being compliant, but there's no clash since there's a custom namespace in front (+ it's meta-data). again my intention was the removal of `tls_peer` hash as we started introducing a convention for plugins to use a prefix convention. the risk of 'breaking' user pipelines initially seemed worth having a clean slate in ECS mode.

not breaking pipelines is more important than being "logically consistent" - **feel free to disregard this review part**.